apply/apply.go

```diff
@@ -43,22 +53,56 @@ func NewApplierFromFile(clusterfile string) (Interface, error) {
 	return NewApplier(cluster)
 }
 
-func GetClusterFromFile(filepath string) (cluster *v1.Cluster, err error) {
-	cluster = &v1.Cluster{}
-	if err = utils.UnmarshalYamlFile(filepath, cluster); err != nil {
-		return nil, fmt.Errorf("failed to get cluster from %s, %v", filepath, err)
-	}
-	cluster.SetAnnotations(common.ClusterfileName, filepath)
-	return cluster, nil
-}
-
 func NewApplier(cluster *v1.Cluster) (Interface, error) {
 	switch cluster.Spec.Provider {
 	case common.AliCloud:
 		return NewAliCloudProvider(cluster)
 	case common.CONTAINER:
 		return NewAliCloudProvider(cluster)
 	}
-
 	return NewDefaultApplier(cluster)
 }
+
+func NewAliCloudProvider(cluster *v1.Cluster) (Interface, error) {
+	k8sClient, err := k8s.Newk8sClient()
+	if err != nil {
+		logger.Warn(err)
```

Comment:
Why not return the non-nil error?

Reply:
here we use k8sClient to determine whether to create a new cluster.Indeed, the logic here can be optimized, i will do it later.